spec/lib/velum/file_config_spec.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+require "tempfile"
+require 'fileutils'
+require "velum/file_config"
+
+describe Velum::FileConfig do
+  conf_dir = File.join(Dir.tmpdir, "velum_config_test")
+
+  before do
+    FileUtils::rm_r(conf_dir, :force => true)
+  end
+
+  after do
+    FileUtils::rm_r(conf_dir, :force => true)
+  end
+
+  it "reads and writes hash structures" do
+    conf = Velum::FileConfig.new(conf_dir)
+    expect(conf.exists?("hash")).to eq false
+
+    conf.write_hash("hash", {"a"=>1})
+    expect(conf.exists?("hash")).to eq true
+    expect(conf.read_hash("hash")).to eq({"a"=>1})
+  end
+
+  it "generates application key in hex string" do
```

Comment:
Description doesn't really match what the test does. Maybe something like:

```generates a unique secret```. I would also wrap this in a "describe #new_secret_key_base" block ([Minor]).